text/entities/0256-entities-data-model.md

```diff
@@ -0,0 +1,790 @@
+    </td>
+   </tr>
+   <tr>
+    <td>Timestamp
+    </td>
+    <td>nanoseconds
+    </td>
+    <td>The time since when the entity state is described by this event.
+The time is measured by the origin clock. The field is required.
+    </td>
+   </tr>
+   <tr>
+    <td>Interval
+    </td>
+    <td>milliseconds
+    </td>
+    <td>Defines the reporting period, i.e. how frequently the
+information about this entity is reported via EntityState events even if
+the entity does not change. The next expected EntityEvent for this
+entity is expected at (Timestamp+Interval) time. Can be used by
+receivers to infer that a no longer reported entity is gone, even if the
+EntityDelete event was not observed. Optional, if missing the interval
+is unknown.
+    </td>
+   </tr>
+   <tr>
+    <td>Type
+    </td>
+    <td>
+    </td>
+    <td>See data model.
+
+</td>
+   </tr>
+   <tr>
+    <td>Id
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+   <tr>
+    <td>Attributes
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+</table>
+
+We say that an entity mutates (changes) when one or more of its
+descriptive attributes changes. A new descriptive attribute may be
+added, an existing descriptive attribute may be deleted or a value of an
+existing descriptive attribute may be changed. All these changes
+represent valid mutations of an entity over time. When these mutations
+happen the identity of the entity does not change.
+
+When the entity's state is changed it is expected that the source will
+emit a new EntityState event with a fresh timestamp and full list of
+values of all other fields.
+
+Entity event producers are recommended to periodically emit events even
+if the entity does not change. In this case the Type, Id and Attribute
+fields will remain the same, but a fresh Timestamp will be recorded in
+the event. Producing such events allows the system to be resilient to
+event losses. Even if some events are lost eventually the correct state
+of the entity is more likely to be delivered to the final destination.
+Periodic sending of EntityState events also serves as a liveliness
+indicator (see below how it can be used in lieu of EntityDelete event).
+
+### EntityDelete Event
+
+EntityDelete event indicates that a particular entity is gone:
+
+<table>
+   <tr>
+    <td><strong>Field</strong>
+    </td>
+    <td><strong>Type</strong>
+    </td>
+    <td><strong>Description</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Timestamp
+    </td>
+    <td>nanoseconds
+    </td>
+    <td>The time when the entity was deleted. The time is measured by
+the origin clock. The field is required.
+    </td>
+   </tr>
+   <tr>
+    <td>Type
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+   <tr>
+    <td>Id
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+</table>
+
+Note that EntityDelete is optional and is not guaranteed to be sent when
```

Comment:
```suggestion
Note that transmitting EntityDelete is voluntary and is not guaranteed to be sent when
```

Reply:
I am not sure the word "voluntary" fits. The intent here is that if the producer is able it absolutely must sent the event, but we know that sometimes the producers won't be able to and that is a valid situation and recipients must be prepared for that. Perhaps I should simply drop the word "optional" and say that it is not guaranteed to be transmitted, e.g.

>Note that transmitting EntityDelete is not guaranteed when

Reply:
Reworded.

Reply:
> Perhaps I should simply drop the word "optional" and say that it is not guaranteed to be transmitted, e.g.
> 
> > Note that transmitting EntityDelete is not guaranteed when

LGTM :+1: